### ISSUE TITLE
Eliot's bug

### ISSUE THREAD

**Issue by donenrique24** [RNCan/BioSimClient_R#9]
Eliot McIntire reported the following bug:

~~~
library(BioSIM)
l <- structure(list(
    Name = c("ID1", "ID2", "ID3", "ID4", "ID5", "ID6"), 
    X = c(-116.623960, -116.459132, -116.294005, 
          -116.128578, -115.962854, -115.796835), 
    Y = c(58.2332800, 
          58.2638898, 58.2942755, 58.3244365, 58.3543720, 
          58.3840815), 
    elev = c(326, 315, 301, 287, 275, 267)), row.names = c(NA, 6L), class = "data.frame")
generateWeather(fromYr = 2020, toYr = 2020,
                l$Name,
                latDeg = l$Y, longDeg = l$X,
                elevM = l$elev,
                modelNames = "MPB_SLR",
                rcp = "RCP85", climModel = "GCM4")

#Starting local Java server...
#If you are using BioSIM Web API for a scientific publication, please cite it. Type citation("BioSIM") in #your R console to obtain the full citation.
#Error in .checkForExceptionInCallback(callback) : 
#  java.lang.NoSuchFieldException: size
#java.lang.Class.getField(Unknown Source)
#j4r.lang.codetranslator.REnvironment.processField(REnvironment.java:384)
#j4r.lang.codetranslator.REnvironment.processCode(REnvironment.java:234)
#j4r.net.server.JavaGatewayServer$JavaGatewayClientThread.processRequest(JavaGatewayServer.java:177)
#j4r.net.server.JavaGatewayServer$JavaGatewayClientThread.run(JavaGatewayServer.java:116)
#java.lang.Thread.run(Unknown Source)
~~~

**Comment by donenrique24** [comment]
It seems the model MPB_SLR requires the MTS from the current and previous years in order to provide the prediction for the current year. Consequently, setting fromYr = 2020 and toYear = 2020 fails. To obtain the prediction for year 2020, the fromYr argument should be set to 2019.
The model help is not explicit on this. It should be improved.

**Comment by eliotmcintire** [comment]
Thank you!

It looks like these models work if there are at least 2 years, but fail with 1 year. Importantly, when supplying more than 1 year, they only return the final year. i.e., if 2010 to 2020 is supplied, it will only return data from 2020.
```
 **[1] "BudBurst"                                            "Climate_Mosture_Index_Annual"                       
 [3] "EmeraldAshBorerColdHardiness_Annual"                 "Gypsy_Moth_Seasonality"                             
 [5] "HemlockWoollyAdelgid_Annual"                         "HemlockWoollyAdelgid_Daily"                         
 [7] "MPB_Cold_Tolerance_Annual"                           "MPB_Cold_Tolerance_Daily"                           
 [9] "MPB_SLR"                                             "PlantHardinessCanada"                               
[11] "PlantHardinessUSA"                                   "Spruce_Budworm_Biology_Annual"                      
[13] "SpruceBeetle"                                        "Standardised_Precipitation_Evapotranspiration_Index"**
```

**Comment by donenrique24** [comment]
You are right with MPB_SLR. However, I just checked with "Climate_Mosture_Index_Annual" because I have been using it recently. When I supply 2015 and 2021 as initial and final years, I get the CMI for all those years except 2015. Same thing for "Gypsy_Moth_Seasonality". 

I'll check the others when I am back to the office.

**Comment by donenrique24** [comment]
These 8 models would not produce a simulation for a single year, and would not provide the initial year in their result:

- BudBurst 
- Climate_Mosture_Index_Annual 
- Gypsy_Moth_Seasonality
- HemlockWoollyAdelgid_Annual
- MPB_Cold_Tolerance_Annual
- MPB_SLR
- Spruce_Budworm_Biology_Annual
- SpruceBeetle

The WebAPI has been adapted to provide this missing initial year. For instance a weather generation for 2003-2008 will produce a value for each year whereas before year 2003 was missing. 

This leaves the following models to check and fix:

- EmeraldAshBorerColdHardiness_Annual
- HemlockWoollyAdelgid_Daily
- MPB_Cold_Tolerance_Daily
- PlantHardinessCanada (this one does not seem to work at all)                               
- PlantHardinessUSA (this one does not seem to work at all)
- Standardised_Precipitation_Evapotranspiration_Index

**Comment by donenrique24** [comment]
After double checking the remaining models, it turns out that 
- PlantHardinessCanada 
- PlantHardinessUSA 
both require meterological time series of 30 years to provide an output. This will not be fixed for now. 

However, the API can be adapted for the other 4 models:
- EmeraldAshBorerColdHardiness_Annual
- HemlockWoollyAdelgid_Daily
- MPB_Cold_Tolerance_Daily
- Standardised_Precipitation_Evapotranspiration_Index

**Comment by donenrique24** [comment]
The WebAPI has been adapted so as to provide output for these models even if the meteorological time series spans on one year:
- EmeraldAshBorerColdHardiness_Annual
- HemlockWoollyAdelgid_Daily
- MPB_Cold_Tolerance_Daily
- Standardised_Precipitation_Evapotranspiration_Index

@eliotmcintire All the changes related to this issues will be enabled with the deployment of the new version of the Web API, which is schedule tomorrow April 10th, if everything goes well.

**Comment by donenrique24** [comment]
MPB_SLR remains an exception. It produces a value for a single year. For instance, if the MTS goes from 2000 to 2010. The MPB_SLR model will produce a single value for 2010.
